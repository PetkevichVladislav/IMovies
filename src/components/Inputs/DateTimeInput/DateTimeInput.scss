.date-time-input{
    width: 100%;
}