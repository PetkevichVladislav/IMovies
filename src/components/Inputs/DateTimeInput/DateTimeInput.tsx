import { FC, useState } from "react";

import "../Input.scss";
import "./DateTimeInput.scss";

export interface IDateTimeInput {
    initialValue?: string;
    name?: string;
}

export const DateTimeInput: FC<IDateTimeInput> = ({ initialValue, name }) => {
    const [date, setDate] = useState<string|undefined>(initialValue);

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setDate(e.target.value);
    };

    return (
        <>
            <input className="input date-time-input" type="date" name={name} onChange={handleChange} value={date} data-testid="date-time-input"/>
        </>
    )
}