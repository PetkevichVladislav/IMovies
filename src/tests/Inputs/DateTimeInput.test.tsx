import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { DateTimeInput, IDateTimeInput } from '../../components/Inputs/DateTimeInput/DateTimeInput';

describe("Date time Input", () => {
    const DateTimeInputData : IDateTimeInput = {
        initialValue: "15-03-2021",
        name: "date",
    };

    it("When data is valid component should renders correctly", () =>{
        const tree = renderer.create(<DateTimeInput {...DateTimeInputData}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("when input is created it should be properly initialized", () =>{
        render(<DateTimeInput {...DateTimeInputData}/>);
        const input = screen.getByTestId("date-time-input");
        expect(input).toHaveAttribute("name",DateTimeInputData.name);
        expect(input).toHaveAttribute("value",DateTimeInputData.initialValue);
    });
});