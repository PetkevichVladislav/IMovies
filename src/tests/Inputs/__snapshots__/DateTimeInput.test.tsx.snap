// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date time Input When data is valid component should renders correctly 1`] = `
<input
  className="input date-time-input"
  data-testid="date-time-input"
  name="date"
  onChange={[Function]}
  type="date"
  value="15-03-2021"
/>
`;
